doctr/utils/geometry.py

```diff
@@ -201,25 +251,27 @@ def rotate_image(
     # Compute the expanded padding
     if expand:
         exp_shape = compute_expanded_shape(image.shape[:-1], angle)
-        h_pad, w_pad = int(math.ceil(exp_shape[0] - image.shape[0])), int(math.ceil(exp_shape[1] - image.shape[1]))
+        h_pad, w_pad = int(max(0, ceil(exp_shape[0] - image.shape[0]))), int(
+            max(0, ceil(exp_shape[1] - image.shape[1])))
         exp_img = np.pad(image, ((h_pad // 2, h_pad - h_pad // 2), (w_pad // 2, w_pad - w_pad // 2), (0, 0)))
     else:
         exp_img = image
 
     height, width = exp_img.shape[:2]
     rot_mat = cv2.getRotationMatrix2D((width / 2, height / 2), angle, 1.0)
-    rot_img = cv2.warpAffine(exp_img.astype(np.float32), rot_mat, (width, height))
+    rot_img = cv2.warpAffine(exp_img, rot_mat, (width, height))
     if expand:
-        # Pad to get the same aspect ratio
-        if (image.shape[0] / image.shape[1]) != (rot_img.shape[0] / rot_img.shape[1]):
-            # Pad width
-            if (rot_img.shape[0] / rot_img.shape[1]) > (image.shape[0] / image.shape[1]):
-                h_pad, w_pad = 0, int(rot_img.shape[0] * image.shape[1] / image.shape[0] - rot_img.shape[1])
-            # Pad height
-            else:
-                h_pad, w_pad = int(rot_img.shape[1] * image.shape[0] / image.shape[1] - rot_img.shape[0]), 0
-            rot_img = np.pad(rot_img, ((h_pad // 2, h_pad - h_pad // 2), (w_pad // 2, w_pad - w_pad // 2), (0, 0)))
-        # rescale
-        rot_img = cv2.resize(rot_img, image.shape[:-1][::-1], interpolation=cv2.INTER_LINEAR)
+        if preserve_origin_shape:
```

Comment:
Sorry I think I skipped that in my last review: the `preserve_origin_shape` should not condition the padding operation (controlled by `expand`), but only the scaling part.

So could you move this "if" right before the scaling part & unindent this block please? :pray:

Reply:
done

Reply:
Yep I did that ! When doing this I realized that the padding was adding much offset. This was because we were using relative boxes with the rotation matrix although here we should be using absolute boxes. I changed that and now it works perfectly.